roxie/ccd/ccdcontext.cpp

```diff
@@ -1903,32 +1903,36 @@ class CRoxieContextBase : implements IRoxieAgentContext, implements ICodeContext
     }
     virtual void getResultDecimal(unsigned tlen, int precision, bool isSigned, void * tgt, const char * stepname, unsigned sequence)
     {
-        if (isSpecialResultSequence(sequence))
+        CriticalBlock b(contextCrit);
+        IPropertyTree &ctx = useContext(sequence);
+        memset(tgt, 0, tlen);
+        if (ctx.hasProp(stepname))
         {
-            MemoryBuffer m;
-            CriticalBlock b(contextCrit);
-            useContext(sequence).getPropBin(stepname, m);
-            if (m.length())
+            if (ctx.isBinary(stepname))
             {
-                assertex(m.length() == tlen);
-                m.read(tlen, tgt);
+                MemoryBuffer m;
+                CriticalBlock b(contextCrit);
```

Comment:
can be removed (and line 1926) because you already hold the critical section.  Can also use ctx instead.